sql/plan/group_by_test.go

```diff
@@ -157,6 +158,54 @@ func TestGroupByAggregationGrouping(t *testing.T) {
 	require.Equal(expected, rows)
 }
 
+func TestGroupByCollations(t *testing.T) {
+	require := require.New(t)
+	ctx := sql.NewEmptyContext()
+
+	colType := sql.MustCreateString(query.Type_VARCHAR, 255, sql.Collation_utf8mb4_0900_ai_ci)
+
+	childSchema := sql.Schema{
+		{Name: "col1", Type: colType},
+		{Name: "col2", Type: sql.Int64},
+	}
+
+	child := memory.NewTable("test", sql.NewPrimaryKeySchema(childSchema), nil)
+
+	rows := []sql.Row{
+		sql.NewRow("col1_1", int64(1111)),
+		sql.NewRow("Col1_1", int64(1111)),
+		sql.NewRow("col1_2", int64(4444)),
+		sql.NewRow("col1_1", int64(1111)),
+		sql.NewRow("Col1_2", int64(4444)),
+	}
+
+	for _, r := range rows {
+		require.NoError(child.Insert(sql.NewEmptyContext(), r))
+	}
+
+	p := NewGroupBy(
+		[]sql.Expression{
+			aggregation.NewSum(
+				expression.NewGetFieldWithTable(1, sql.Int64, "test", "col2", false),
+			),
+		},
+		[]sql.Expression{
+			expression.NewGetFieldWithTable(0, colType, "test", "col1", false),
+		},
+		NewResolvedTable(child, nil, nil),
+	)
+
+	rows, err := sql.NodeToRows(ctx, p)
+	require.NoError(err)
+
+	expected := []sql.Row{
+		{float64(3333)},
+		{float64(8888)},
+	}
+
+	require.Equal(expected, rows)
```

Comment:
Let's add a few additional tests for `ENUM` and `SET` as well. I'd expect them to work as, although they're collation-aware, they handle their values as integers, so I'd expect it to just work. But we should verify the behavior anyway.